src/components/text-input/_text-input.scss

```diff
@@ -90,3 +90,87 @@
     }
   }
 }
+
+@mixin text-input--experimental {
+  .#{$prefix}--text-input {
+    @include reset;
+    @include typescale('zeta');
+    background-color: $field-01;
+    width: 100%;
+    height: rem(40px);
+    padding: 0 $spacing-xl 0 $spacing-md;
+    color: $text-02;
+    border: none;
+    outline: 1px solid $ui-04;
+    outline-offset: -1px;
+  }
+
+  .#{$prefix}--text-input ~ .#{$prefix}--label {
+    @include typescale('omega');
+    padding: 0;
+    font-weight: 400;
+    display: block;
+    padding: $spacing-sm 0 $spacing-3xl 0;
+    margin: 0;
+  }
+
+  .#{$prefix}--text-input:required ~ .#{$prefix}--label::after {
+    content: ' *';
+  }
+
+  .#{$prefix}--text-input::-webkit-input-placeholder {
+    color: $text-03;
+  }
+
+  .#{$prefix}--text-input:hover {
+    color: $text-01;
+    outline: 1px solid $brand-01;
+    outline-offset: -1px;
+  }
+
+  .#{$prefix}--text-input:focus,
+  .#{$prefix}--text-input:active {
+    outline: 2px solid $brand-01;
+    outline-offset: -2px;
```

Comment:
Seems like `IE11` doesn't support outline-offset, so I would just make sure this looks okay in that browser (My VM is not working atm 😢 )

Reply:
oh good call. will check 😞

Reply:
It just shows the outline on the outside instead of the inside, theres not really an easy way to adjust this so I think its fine. 
<img width="360" alt="screen shot 2018-09-19 at 11 56 50 am" src="https://user-images.githubusercontent.com/2753488/45768656-1fd86800-bc03-11e8-8951-eb39cb3dcfef.png">

Reply:
FIne by me!